doc/handle_scope.md

```diff
@@ -1,5 +1,67 @@
-# Handle scope
+# HandleScope
 
-You are reading a draft of the next documentation and it's in continuos update so
-if you don't find what you need please refer to:
-[C++ wrapper classes for the ABI-stable C APIs for Node.js](https://nodejs.github.io/node-addon-api/)
\ No newline at end of file
+The HandleScope class is used to manage the lifetime of object handles
+which are created through the use of node-addon-api. These handles
+keep an object alive in the heap in order to ensure that the objects
+are not collected while native code is using them.  A handle may be
+created any time a node-addon-api Value(and subclasses) are created
+or returned.  
```

Comment:
A handle may be created when any new node-addon-api Value and its subclasses have been created or returned.